js/chat/chat.ts

```diff
@@ -262,6 +269,7 @@ class ChatInput extends LightElement {
 }
 
 class ChatContainer extends LightElement {
+  @property() icon = "";
```

Comment:
This property should be `iconAssistant` since we'll likely end up with customizations for other icons. I initially wrote this feature using attributes in this same way, and I'm glad you resolved the dependency issue (TIL `.get_dependencies()` method!) but I still have a small reservation that we only use this attribute on connected callback and then ignore it. And I guess there's some part of this that just doesn't sit well with me that we're relying on attributes when HTML and the DOM are right there.

Reply:
I went _yet another way_ with this in 048a36b8176327d2f0794cd75a6f2a57be3f4973 and updated `<shiny-chat-container>` and `<shiny-chat-message>` to use the shadow DOM, allowing us to use slots for the icon.

This makes the components slightly more complicated -- some of the icon styles move into the component for example -- but it's web-native and consistent.

Reply:
The shadow DOM + slots approach will require some test updates, but I wanted to pause for feedback before updating those.

Reply:
Ah yeah, it breaks syntax highlighting because the `<shiny-markdown-stream>` element ends up in the ShadowDOM, but I think we can work around this, or maybe go back to the light DOM for `<shiny-chat-message>`

Reply:
For the sake of feeling spread thin on a lot of different things right now, I'd prefer to revert back to https://github.com/posit-dev/py-shiny/commit/048a36b8176327d2f0794cd75a6f2a57be3f4973 and call this PR done.

Reply:
I fixed it in da8803cf759bf704c3bd79991d6158b90fb85d9e

Reply:
...and back to attributes in f0ac89d204f245c2f181d7591ba64cdcb18dcafd